src/matvis/cpu/coords.py

```diff
@@ -0,0 +1,215 @@
+"""Coordinate rotation methods."""
+
+import numpy as np
+from astropy import units as un
+from astropy.coordinates import AltAz
+from astropy.coordinates.erfa_astrom import erfa_astrom
+
+from ..coordinates import point_source_crd_eq
+from ..core.coords import CoordinateRotation
+
+# Schwarzschild radius of the Sun (au) */
+ERFA_SRS = 1.97412574336e-8
+
+
+class CoordinateRotationAstropy(CoordinateRotation):
+    """Perform coordinate rotation with astropy directly."""
+
+    def setup(self):
+        """Warmup cache as well as initializing memory."""
+        super().setup()
+
+        # Do one rotation to warm up the cache.
+        frame = AltAz(obstime=self.times[0], location=self.telescope_loc)
+        self.skycoords[0].transform_to(frame)
+
+    def rotate(self, t: int) -> tuple[np.ndarray, np.ndarray]:
+        """Compute the topocentric coordinates from the given time and telescope loc."""
+        frame = AltAz(obstime=self.times[t], location=self.telescope_loc)
+        altaz = self.skycoords.transform_to(frame)
+        el, az = self.xp.asarray(altaz.alt.rad), self.xp.asarray(altaz.az.rad)
+
+        # Astropy has Az oriented East of North, i.e. Az(N) = 0 deg, Az(E) = +90 deg.
+        self.all_coords_topo[0] = self.xp.cos(el) * self.xp.sin(az)
+        self.all_coords_topo[1] = self.xp.cos(el) * self.xp.cos(az)
+        self.all_coords_topo[2] = self.xp.sin(el)
+
+        if self.gpu:
+            self.xp.cuda.Device().synchronize()
+
+
+class CoordinateRotationERFA(CoordinateRotation):
```

Comment:
Can we add references to astropy/ERFA functions, for which the corrections were based on?

Reply:
I can.